src/main/kotlin/no/nav/omsorgsdageraleneomsorgapi/søknad/SøknadService.kt

```diff
@@ -28,11 +32,35 @@ class SøknadService(
 
         val barnMedIdentitetsnummer = barnService.hentNåværendeBarn(idToken, callId)
         søknad.oppdaterBarnMedIdentitetsnummer(barnMedIdentitetsnummer)
-        søknad.valider()
 
+        søknad.valider()
         LOGGER.info(formaterStatuslogging(søknad.søknadId, "validert OK"))
 
-        kafkaProducer.produserKafkamelding(søknad = søknad.tilKomplettSøknad(søker), metadata = metadata)
+        if (søknad.barn.size > 1) {
+            val søknader = søknad.splittTilSøknadPerBarn()
+            LOGGER.info("SøknadId:${søknad.søknadId} splittet ut til ${søknader.map { it.søknadId }}")
+            try {
+                kafkaProdusent.beginTransaction()
+                søknader.forEach { kafkaProdusent.produserKafkamelding(søknad = it.tilKomplettSøknad(søker), metadata = metadata) }
+                kafkaProdusent.commitTransaction()
+            } catch (e: Exception) {
+                LOGGER.error("Feilet ved produsering av kafkamelding")
+                when (e) {
+                    is ProducerFencedException, is OutOfOrderSequenceException, is AuthorizationException -> {
+                        // We can't recover from these exceptions, so our only option is to close the producer and exit.
+                        kafkaProdusent.close()
```

Comment:
Det er ikke så lurt å kalle på close() ved feil. 
Dette vil føre til at kafkaProducer lukkes ved en slik feil, og kan kun starte ved en restart av tjenesten.

Reply:
Okei, tok det fra docen. https://kafka.apache.org/25/javadoc/org/apache/kafka/clients/producer/KafkaProducer.html
Altså ikke gjøre noe, bare kaste unntaket videre til klienten også prøve igjen ved ny innsending?

Reply:
Okei kan prøve det :)

Reply:
Tenker du på producer.close(); i doken?

Reply:
Ja altså feilhåndteringen er tatt derifra, men være forskjellig case

Reply:
Den er lagt på slutten av program for å indikere å lukke producer etter endt program (altså ikke funksjonakall, men når applikasjonen skrus av). Du burde legge dette kallet til i en shutdown lister på ktor, lik det vi har gjort på andre proesseringstjenster for clearing av CollectorRegistry.

Reply:
Aha sånn ja. Okei, fikser det :D

Reply:
Fikset